src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -918,12 +1010,12 @@ private int decodeSaslInit(
         final int limit)
     {
         int progress;
-        final AmqpPerformativeFW performative = amqpPerformativeRO.wrap(buffer, offset, limit);
-        final AmqpSaslInitFW saslInit = performative.saslInit();
+        final AmqpSecurityFW security = amqpSecurityRO.wrap(buffer, offset, limit);
+        final AmqpSaslInitFW saslInit = security.saslInit();
         assert saslInit != null;
 
         server.onDecodeSaslInit(traceId, authorization, saslInit);
-        server.decoder = decodeProtocolHeader;
+        server.decoder = decodeProtocolHeaderZero;
```

Comment:
This direct assignment to `decodeProtocolHeaderZero` assumes that the `sasl` handshake is over when `onDecodeSaslInit` returns.

However, even though that is true for the current simplified case, where the `sasl` `outcome` is produced immediately by `onDecodeSaslInit`, that is not the general case where multiple `sasl` frames must interplay back and forth before the `sasl` handshake is complete.

Therefore, it might be better to track the `sasl` `outcome` in `onDecodeSaslInit` just prior to calling `doEncodeSaslOutcome`, have this `decodeSaslInit` always set the decoder back to `decodeSaslFrame` and then in `decodeSaslFrame`, check for `sasl` `outcome`, deferring to `decodeProtocolHeaderZero` if `sasl` handshake is complete.

That subtle change in logic should address both this simplified `sasl` handshake and also the more general case later.